vertica_python/vertica/cursor.py

```diff
@@ -58,8 +58,7 @@ def execute(self, operation, parameters=None):
 
         self.rowcount = 0
         if self.last_execution:
-            # ToDo: can just empty the message buffer in connection if easy to do
-            self.connection.reset_connection()
```

Comment:
OK. A cleaner way to support open transactions is:
(i) In `execute()`, if last_execution is not null, read/drop messages until ReadyForQuery
(ii) Keep `fetchone()` light weight; `fetchone()` shouldn't do too much processing beyond its scope.